graylog2-web-interface/src/components/navigation/MainNavbar.tsx

```diff
@@ -157,19 +78,29 @@ const pluginLicenseValid = (navigationItems: Array<PluginNavigation>, descriptio
   return menuItem && Object.keys(menuItem).includes('useIsValidLicense') ? menuItem.useIsValidLicense() : true;
 };
 
-const sortItemsByPosition = <T extends { position: typeof LAST_POSITION | undefined }>(navigationItems: Array<T>) =>
-  navigationItems.sort((route1, route2) => {
-    if (route1.position === LAST_POSITION) {
-      return 1;
-    }
+const sortInAfterItem = (targetList: Array<PluginNavigation>, afterItems: Array<PluginNavigation>) => {
+  const result = [...targetList];
 
-    if (route2.position === LAST_POSITION) {
-      return -1;
+  afterItems.forEach((afterItem) => {
+    const index = result.findIndex((targetItem) => targetItem.description === afterItem.position?.after);
+    if (index !== -1) {
+      result.splice(index + 1, 0, afterItem);
+    } else {
+      result.push(afterItem);
     }
-
-    return 0;
   });
 
+  return result;
```

Comment:
I considered using `reduce` here, for a more functional approach, but in my opinion it is harder to understand.

```
const sortInAfterItem = (targetList: Array<PluginNavigation>, afterItems: Array<PluginNavigation>) =>
  afterItems.reduce(
    (result, item) => {
      const index = result.findIndex((sortedItem: PluginNavigation) => sortedItem.description === item.position?.after);
      if (index !== -1) {
        result.splice(index + 1, 0, item);
      } else {
        result.push(item);
      }

      return result;
    },
    [...targetList],
  );

```